commandsv3/src/main/java/org/wpilib/commands3/RequireableResource.java

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package org.wpilib.commands3;
+
+import edu.wpi.first.units.measure.Time;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * A resource that may be claimed by a command. A single claimable resource cannot be claimed by
+ * more than one running command at a time.
+ */
+public class RequireableResource {
```

Comment:
This has going through several rounds of renaming over the last year+, and I'm honestly not a fan of any of the variations on "resource" or "requireable". It's so generic and doesn't really cover what most people use this for: representing a mechanism in code. `Mechanism` is too far in the other direction, since it could be an entire mechanism (like an arm) or a controllable subassembly (like a single joint on an arm).

I think `Subsystem` might still be the best choice here.

Reply:
"Subsystem" has been used since the original commands framework came out in 2012, so all examples and team code already use it. But it's also less proscriptive than "Mechanism"; subsystems don't need to be mechanical, like LED strips or cameras that can be switched between different processing modes.